### PR TITLE
bh/schema sub commands

### PR DESCRIPTION
**Pull request by bheni**
This is a refactor of the existing dolt schema functionality.  Previously dolt schema had many different ways it could be run:

```
        "[<commit>] [<table>...]",
	"--export <table> <file>",	
	"--add-column [--default <default_value>] [--not-null] [--tag <tag-number>] <table> <name> <type>",	
	"--rename-column <table> <old> <new>",	
	"--drop-column <table> <column>",
```

These have all been split into separate commands:

dolt schema show
dolt schema export
dolt schema add-column
dolt schema rename-column
dolt schema drop-column

### CODE REVIEW

File: go/cmd/dolt/commands/schcmds/drop_column.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+var schDropColShortDesc = "Removes a column of the specified table"
+var schDropColLongDesc = ""
+var schDropColSynopsis = []string{
+	"<table> <column>",
+}
+
+func DropColumn(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv) int {
+	ap := argparser.NewArgParser()
+	ap.ArgListHelp["table"] = "table(s) whose schema is being displayed."
```
Comment (zachmu) [inline go/cmd/dolt/commands/schcmds/drop_column.go]: Need updated help text

File: go/cmd/dolt/commands/schcmds/add_column.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+var schAddColShortDesc = "Adds a column to specified table's schema"
+var schAddColLongDesc = "Adds a column to the specified table's schema. If no default value is provided the column will be empty"
+var schAddColSynopsis = []string{
+	"[--default <default_value>] [--not-null] [--tag <tag-number>] <table> <name> <type>",
+}
+
+func AddColumn(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv) int {
+	ap := argparser.NewArgParser()
+	ap.ArgListHelp["table"] = "table(s) whose schema is being displayed."
```
Comment (zachmu) [inline go/cmd/dolt/commands/schcmds/add_column.go]: Need to update help text

File: go/cmd/dolt/commands/schcmds/export.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+var schExportShortDesc = "Exports a table's schema."
+var schExportLongDesc = ""
+var schExportSynopsis = []string{
+	"<table> <file>",
+}
+
+func Export(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv) int {
+	ap := argparser.NewArgParser()
+	ap.ArgListHelp["table"] = "table(s) whose schema is being displayed."
```
Comment (zachmu) [inline go/cmd/dolt/commands/schcmds/export.go]: Update help text

File: go/cmd/dolt/commands/schcmds/rename_column.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+var schRenameColLongDesc = "<b>dolt schema rename-column</b> will change the display name of a column. Changing the name" +
+	"of the column will only modify the schema and no data will change."
+var schRenameColSynopsis = []string{
+	"<table> <old> <new>",
+}
+
+func RenameColumn(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv) int {
+	ap := argparser.NewArgParser()
+	ap.ArgListHelp["table"] = "table being modified."
+	ap.SupportsFlag(notNullFlag, "", "If provided rows without a value in this column will be considered invalid.  If rows already exist and not-null is specified then a default value must be provided.")
```
Comment (zachmu) [inline go/cmd/dolt/commands/schcmds/rename_column.go]: This flag shouldn't apply to a rename

File: go/cmd/dolt/commands/schcmds/schema.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"github.com/liquidata-inc/dolt/go/cmd/dolt/cli"
+	eventsapi "github.com/liquidata-inc/dolt/go/gen/proto/dolt/services/eventsapi/v1alpha1"
+)
+
+var Commands = cli.GenSubCommandHandler([]*cli.Command{
+	{Name: "add-column", Desc: "Adds a column to specified table's schema.", Func: AddColumn, ReqRepo: true, EventType: eventsapi.ClientEventType_SCHEMA},
+	{Name: "drop-column", Desc: "Removes a column of the specified table.", Func: DropColumn, ReqRepo: true, EventType: eventsapi.ClientEventType_SCHEMA},
+	{Name: "export", Desc: "Exports a table's schema.", Func: Export, ReqRepo: true, EventType: eventsapi.ClientEventType_SCHEMA},
+	//{Name: "import", Desc: "Creates a new table with an inferred schema.", Func: Import, ReqRepo: true, EventType: eventsapi.ClientEventType_SCHEMA},
```
Comment (zachmu) [inline go/cmd/dolt/commands/schcmds/schema.go]: Commented out on purpose? You have an implementation

Comment (bheni) [reply go/cmd/dolt/commands/schcmds/schema.go]: yes, out on purpose.  Will be fully implemented in a separate PR.

File: go/cmd/dolt/commands/schcmds/rename_column.go (hunk 4 above)
Comment (bheni) [reply go/cmd/dolt/commands/schcmds/rename_column.go]: fixed

File: go/cmd/dolt/commands/schcmds/export.go (hunk 3 above)
Comment (bheni) [reply go/cmd/dolt/commands/schcmds/export.go]: updated

File: go/cmd/dolt/commands/schcmds/drop_column.go (hunk 1 above)
Comment (bheni) [reply go/cmd/dolt/commands/schcmds/drop_column.go]: updated

File: go/cmd/dolt/commands/schcmds/add_column.go (hunk 2 above)
Comment (bheni) [reply go/cmd/dolt/commands/schcmds/add_column.go]: updated

**Pull request merged**